src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -170,5 +177,318 @@ class C
+            var source = @"
+class C
+{
+    void M(string name! = null) { }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var m = comp.GlobalNamespace.GetTypeMember("C").GetMember<SourceMethodSymbol>("M");
+            Debug.Assert(((SourceParameterSymbol)m.Parameters[0]).IsNullChecked);
+        }
+
+        [Fact]
+        public void NullCheckedMethodValidationWithOptionalStringLiteralParameter()
+        {
+            var source = @"
+class C
+{
+    void M(string name! = ""rose"") { }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var m = comp.GlobalNamespace.GetTypeMember("C").GetMember<SourceMethodSymbol>("M");
+            Debug.Assert(((SourceParameterSymbol)m.Parameters[0]).IsNullChecked);
+        }
+
+        [Fact]
+        public void NullCheckedMethodValidationWithOptionalParameterSplitBySpace()
+        {
+            var source = @"
+class C
+{
+    void M(string name! =null) { }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var m = comp.GlobalNamespace.GetTypeMember("C").GetMember<SourceMethodSymbol>("M");
+            Debug.Assert(((SourceParameterSymbol)m.Parameters[0]).IsNullChecked);
+        }
+
+        [Fact]
+        public void NullCheckedConstructor()
+        {
+            var source = @"
+class C
+{
+    public C(string name!) { }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var m = comp.GlobalNamespace.GetTypeMember("C").GetMember<SourceMethodSymbol>(".ctor");
+            Debug.Assert(((SourceParameterSymbol)m.Parameters[0]).IsNullChecked);
+        }
+
+        [Fact]
+        public void NullCheckedOperator()
+        {
+            var source = @"
+class Box 
+{ 
+    public static int operator+ (Box b!, Box c)  
+    { 
+        return 2;
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var m = comp.GlobalNamespace.GetTypeMember("Box").GetMember<SourceUserDefinedOperatorSymbol>("op_Addition");
+            Debug.Assert(((SourceParameterSymbol)m.Parameters[0]).IsNullChecked);
+            Debug.Assert(!((SourceParameterSymbol)m.Parameters[1]).IsNullChecked);
+        }
+
+        [Fact]
+        public void NullCheckedLambdaParameter()
+        {
+            var source = @"
+using System;
+class C
+{
+    public void M()
+    {
+        Func<string, string> func1 = x! => x + ""1"";
+    }
+}";
+            var tree = Parse(source);
+            var comp = CreateCompilation(tree);
+            comp.VerifyDiagnostics();
+
+            var model = (CSharpSemanticModel)comp.GetSemanticModel(tree);
+            var m = comp.GlobalNamespace.GetTypeMember("C").GetMember<SourceMethodSymbol>("M");
```

Comment:
Unused.

Reply:
m is used to define node, and model + node are used to define lambdaSymbol. Which var is unused?

Reply:
Ah, you don't actually need to calculate `m` there, you can just call `tree.GetRoot().DescendentNodes()...` etc